[18.0][MIG] RMA

JasminSForgeFlow opened the pull request:
Depends on:

- https://github.com/OCA/stock-logistics-workflow/pull/1880

Improve tests for 2-step pickings
- update legacy way routes
- add a new test with the new flow of Odoo 18 for 2 steps picking

Fixed Qty to Received calculation and related tests

JordiBForgeFlow committed:
init branch

LoisRForgeFlow committed:
[9.0] rma: remove rma.rule and add that setting to product.category

LoisRForgeFlow committed:
[9.0][FIX] rma:
* fix assignment of moves.
* default qty in rma lines.
* remove account dependency.
* test and flake8 fixes.

AaronHForgeFlow committed:
[IMP] default operation in product and product_categ for customer and supplier
[IMP]Separate menus for customer and supplier operations
* Add active field to rma operation
* Added tests
* Fix travis
* Fix create supplier rma from customer rma

LoisRForgeFlow committed:
[9.0][FIX]
* rma: receipt_policy selections not matching.
* rma_sale: fix _prepare_rma_line_from_sale_order_line.

LoisRForgeFlow committed:
[9.0][IMP] rma_account:
* remove unneded copy attributes.
* simplify action_view methods.
* fix wrong naming.
* fix misplaced views.
* fix wrong count and view actions for rma.orders in invoices.
* fix error when installing the module.
* remove unneded data update when preparing rma lines from invoice lines.
* minor extra fixes.

LoisRForgeFlow committed:
[9.0][IMP] rma_purchase:
* remove unneded copy and ondelete attributes.
* simplify action_view methods.
* fix rma line supplier view.
* fix wizard.
* extend README.
* minor extra fixes.

LoisRForgeFlow committed:
[9.0][REW] rma: workflow centralized on rma.order.line and the use of rma.order is optional.

LoisRForgeFlow committed:
[9.0][IMP] rma: add constrains

LoisRForgeFlow committed:
[9.0][FIX] rma: wizard

LoisRForgeFlow committed:
fix rma

LoisRForgeFlow committed:
[9.0][IMP] rma: add button to rma's from customers

LoisRForgeFlow committed:
add partner constrain

LoisRForgeFlow committed:
[9.0] add under_warranty field

LoisRForgeFlow committed:
[9.0][FIX] rma: create supplier rma wizard

nikul-serpentcs committed:
[IMP] Improved Unit Test Case and Fixed Travis

AaronHForgeFlow committed:
[MIG]rma v10

AaronHForgeFlow committed:
[FIX]view

AaronHForgeFlow committed:
[FIX]various fixes

nikul-serpentcs committed:
[MIG] Migrated UT & Fixed Travis

AaronHForgeFlow committed:
[FIX]error in compute method

nikul-serpentcs committed:
[IMP] Improved Code.

AaronHForgeFlow committed:
[MIG]rma_operating_unit to v10

AaronHForgeFlow committed:
[FIX]moved_qty uses moves not procurements

AaronHForgeFlow committed:
[FIX]compute qty when stock moves but not procurements

Commit:
[11.0] MIG: RMA module

max3903 committed:
[FIX] Permissions and remove (en)coding

max3903 committed:
[MIG] Migrate configuration and cleanup

max3903 committed:
[FIX] Errors from tests

JordiBForgeFlow committed:
multiple fixes

Commit:
[IMP] Various improvements

LoisRForgeFlow committed:
[9.0][IMP] rma: tests moved to Savedpointcase and optimized (66% time reduction)

AaronHForgeFlow committed:
[FIX]RMA location company is the warehouse company

AaronHForgeFlow committed:
[IMP]add rma line report

max3903 committed:
[FIX] Cannot create partner

max3903 committed:
[FIX] Add read access to stock.move

max3903 committed:
[FIX] Add read access to stock.move

max3903 committed:
[FIX] Access rights with group deps

max3903 committed:
[FIX] Access rights with group deps

Commit:
[FIX] TypeError: unhashable type: 'list'

AaronHForgeFlow committed:
[MIG]rma v110.0.2.0.0

AaronHForgeFlow committed:
[FIX]views rma

AaronHForgeFlow committed:
[FIX]compute_qty_to_receive

AaronHForgeFlow committed:
[FIX]rma test

AaronHForgeFlow committed:
[IMP]product first on view then select document according to product

AaronHForgeFlow committed:
[IMP] add group to manage rma groups

LoisRForgeFlow committed:
[9.0][IMP] rma: configuring rma for a warehouse from the wh form view

AaronHForgeFlow committed:
[IMP]add icon. Courtesy lreficent.

AaronHForgeFlow committed:
[FIX]text on create supplier rma button

AaronHForgeFlow committed:
[UPT]report usable in v11

AdriaGForgeFlow committed:
[12.0][MIG] Migrate rma module to v12.0

AdriaGForgeFlow committed:
[12.0][FIX] Fix search view in rma module

bjeficent committed:
[FIX] qty_delivered is not updated properly

grindtildeath committed:
rma: Fix stock_location res.groups

grindtildeath committed:
Proxy fields defaults with lambda to allow inheritance

grindtildeath committed:
Use strings on fields compute to allow inheritance

grindtildeath committed:
Define default value for required fields

grindtildeath committed:
[FIX] Remove picking_id from default_get as model does not define it

Commit:
[SET] Correct website URL for RMA modules

Commit:
[FIX] Class typo, ProcurementRule is StockRule in V12.0

gurneyalex committed:
[FIX] multicompany security

the rma module had company_id fields on rma.order and rma.order.line but not on rma.operation,
and no global multicompany record rules.

-> we fix this and avoid a mess with people allowed to see an operation in the drop down list
but unable to use it because if points to a warehouse of another company

gurneyalex committed:
[IMP] rma operation form view

use the archive button in the form view, rather than
a normal boolean field.

gurneyalex committed:
[FIX] wrong company on stock rules

when working with odoobot account, you would get the wrong company_id
on the RMA stock.rules of the warehouse on which you enabled RMA

gurneyalex committed:
make the RMA routes shared in multicompany by default

AaronHForgeFlow committed:
[FIX]do not copy name when duplicating

AaronHForgeFlow committed:
[FIX]consistency group vs line in the picking count methods

AaronHForgeFlow committed:
[FIX]wizard to create pickings, not to show rma groups if rma groups are not activated

AaronHForgeFlow committed:
[ENH]activate description also fro customer RMAs
[ENH]add term and conditions field to be printed and send to the partner
[FIX]do not show rma groups on the rma line form view if they are not activated

AaronHForgeFlow committed:
[FIX]supplier lines field description

AaronHForgeFlow committed:
[IMP]return qty instead of ordered qty

AaronHForgeFlow committed:
[FIX] description on rma models

AaronHForgeFlow committed:
[FIX]move constraint from the rma order to the line

AaronHForgeFlow committed:
[FIX]UsreError to Validation Error

MiquelRForgeFlow committed:
[FIX] tests

MiquelRForgeFlow committed:
[FIX] default_gets: avoid using shadowname 'fields'

chafique-delli committed:
[10.0][IMP]add report for rma group

AaronHForgeFlow committed:
[UPT]rma group report to v12

AaronHForgeFlow committed:
[IMP] : black, isort

AaronHForgeFlow committed:
[MIG] rma: Migration to 13.0

HviorForgeFlow committed:
RMA as an APP

AaronHForgeFlow committed:
[ENH]rma_sale traceability

AaronHForgeFlow committed:
[IMP]rma master data menus

AaronHForgeFlow committed:
[FIX]rma_account refund creation

AaronHForgeFlow committed:
[IMP]rma enable invoicing from settings

AaronHForgeFlow committed:
[FIX]rma report templates group

HviorForgeFlow committed:
[IMP] Update pre-commit lints according with OCA ones

Commit:
[FIX] External ID not found in the system: account.res_partner_action_supplier

MateuGForgeFlow committed:
[IMP] rma: black, isort, prettier

MateuGForgeFlow committed:
[MIG] rma: Migration to 14.0

MateuGForgeFlow committed:
[IMP] rma
Improve security

MateuGForgeFlow committed:
[MIG] rma: Migration to 14.0 - fix

MateuGForgeFlow committed:
[MIG] rma: Migration to 14.0 - fix

MateuGForgeFlow committed:
[MIG] rma: Migration to 14.0 - fix

MateuGForgeFlow committed:
[MIG] rma: Migration to 14.0 - fix

MateuGForgeFlow committed:
[FIX] rma: Create rma supplier action

MateuGForgeFlow committed:
[MIG] rma_account: Migration to 14.0

MateuGForgeFlow committed:
[14.0][IMP]Fill rma lines in tree (#156)

* [IMP] rma: Fill rma lines in tree

RLeeGML committed:
[14.0][FIX] warehouse rma location

LoisRForgeFlow committed:
[FIX] rma: toggle archive button removed in favor of web_ribbon.

LoisRForgeFlow committed:
[IMP] rma: Do not highlight all action buttons and simplify
rma.line views, supplier view is an extension of customer one.

This makes easier to edit common elements in both views and ease
maintenance.

LoisRForgeFlow committed:
[14.0][MIG] rma*: ir.actions.act_window has different access
right in v14. Actions that read those records need to use
`sudo` to allow non-admin users to be able to use these actions.

MateuGForgeFlow committed:
Fix Pre-commit Websites

JordiBForgeFlow committed:
[rma][fix] allow to set back to approved a done rma

JasminSForgeFlow committed:
[MIG] rma: Migration to 15.0

AaronHForgeFlow committed:
[14.0][IMP]rma: make operation editable after approved
* remove also validation error when setting the rma to draft where there are done pickings

ChrisOForgeFlow committed:
[IMP] rma: Added serial/lot selection on from stock move wizard on rma groups
- added restriction to approve rma with product tracking on serial, should be only one to receive

ChrisOForgeFlow committed:
[IMP] rma, rma_sale: added select all, messages for UX

JordiBForgeFlow committed:
[fix] rma: when using 2 step receipt or delivery, don't count double

JordiBForgeFlow committed:
[fix] rma: improve logic to count in and out pickings
In the scenario where we use a 2 or 3 step receipts or pickings
we want to make sure that we correctly count and classify the pickings.

JordiBForgeFlow committed:
[FIX] rma: compute out_shipment_count correctly. Add test cases

ChrisOForgeFlow committed:
[FIX] rma: add lot to pickings created from wizard on RMA lines

ChrisOForgeFlow committed:
[15.0][ADD] Mass action for request approval RMA order line

JasminSForgeFlow committed:
[IMP] rma: Fix Route View

JordiBForgeFlow committed:
[IMP] rma: Refactor all rma modules in order to consider using the correct price unit in moves
Otherwise the inventory accounting will be completely wrong.

ChrisOForgeFlow committed:
[14.0][FIX] rma: separate stock.move by rma_line_id to fix picking association

AaronHForgeFlow committed:
[15.0][IMP] Tests for stock valuation
[FIX] rma: rma_custmer_user has no write permissions in partner, so compute method fails.
[IMP] rma: use rma user in tests
[FIX] rma_account: move_line_id field string
[IMP] rma, rma_account, rma_sale, rma_purchase: tests for stock valuation
[FIX] account_move_line_rma_order_line: minor lint, make auto-install

LoisRForgeFlow committed:
[FIX] rma: do not force reservation on supplier RMA deliveries

* remove code call to unexisting `force_asign`.
* only one model per model file and rename as appropriate.

astirpe committed:
[13.0][FIX/IMP] Make RMA Operation settings company dependent

AaronHForgeFlow committed:
[IMP] COPIER UPDATE: black, isort, prettier

florian-dacosta committed:
Add missing fields in rma line editable tree in rma group

florian-dacosta committed:
[FIX] Remove useless/bad wiz_id field in rma_make_picking.wizard item preparation

florian-dacosta committed:
[FIX] count produced products going to customer as out pickings

LoisRForgeFlow committed:
[IMP] rma: add RMA reference to delivery slip report

DavidFIB committed:
[14.0][FIX/IMP] Make RMA Operation settings company dependent

DavidFIB committed:
[15.0][FIX/IMP] Make RMA Operation settings company dependent

LoisRForgeFlow committed:
[IMP] rma: add rma lines to group selecting serial numbers

LoisRForgeFlow committed:
[FIX] rma:  missing migration scripts and version bump

Those are needed after making some fields company dependent

florian-dacosta committed:
Hide smart button when empty

StefanRijnhart committed:
[IMP] rma: prevent the creation of zero qty moves

StefanRijnhart committed:
[FIX] rma: prevent against warehouse mismatch or missing rules

When creating pickings, ensure that the applied stock rule was taken from
the operation's routes. Otherwise, the default procurement rules for a
warehouse may kick in, creating incoming customer goods not from the customer
location but from the resupply warehouse.

Commit:
[IMP] Improve multi-company record rules

DavidJForgeFlow committed:
[FIX]rma: remove test_rma dependency to Account

StefanRijnhart committed:
[FIX] rma: improve check on rule selection during procurement

Fixes #274

Thanks to @florian-dacosta for suggesting this approach.

JordiBForgeFlow committed:
[IMP] rma_sale: introduce new config settings.
- auto_confirm_rma_sale
- free_of_charge_rma_sale

florian-dacosta committed:
[IMP] Make rma order view cleaner for user
Hide button and fields depending on the policy chosen on the rma line

DavidJForgeFlow committed:
[15.0][FIX] rma_sale: fixup of procurement

AnnaPForgeFlow committed:
[IMP] rma: add translations

JordiBForgeFlow committed:
[FIX] include anglo-saxon price unit calculation in refunds.
Otherwise the anglo saxon entries won't be correct.
For example, the Interim (Delivered) account should balance
after receiving and triggering a refund on a customer rma.

JordiBForgeFlow committed:
[IMP] centralize the logic to get the correct cost of the RMA.

JordiBForgeFlow committed:
[FIX] fix rma management using multi step routes

florian-dacosta committed:
[16.0][MIG] Migrate rma module to v16

DavidJForgeFlow committed:
[14.0][IMP] rma: pass product in create lot

JordiBForgeFlow committed:
[FIX] fix empty parter on rma picking

florian-dacosta committed:
[FIX] Move some field from onchange to compute fields to avoid issues in views

DavidJForgeFlow committed:
[14.0][IMP] rma: add lot/serial in report

LoisRForgeFlow committed:
[FIX] rma: Ensure that configuration on the operation is applied

Without this, some policies are not being copied from the
operation selected when creating new rma line from a rma group.

In v16 this patch and the usage of such onchange can be removed
in favor of (pre)computed stored editable fields for all policies
and configuration in the RMA operation.

LoisRForgeFlow committed:
[IMP] rma: rma group supplier form view as a inherited view.

The goal is to simplify view defintion and do not duplicate
things, requiring to do view changes in two places.

The same change was done from rma lines some time ago.

LoisRForgeFlow committed:
[IMP] rma: add description to rma order and copy it over to new lines

LoisRForgeFlow committed:
[IMP] rma: order by id desc

LoisRForgeFlow committed:
[IMP] rma: print serial/lot number in RMA group report

LoisRForgeFlow committed:
[FIX] rma: remove "Add new line" in RMA group.

This was not the intended way to add RMA lines to a group.
Users are supposed to use the wizards to do so (add from stock
move, add from serial, etc.). Having the option to use "add
new line" was only leading to errors and confusion.

DavidJForgeFlow committed:
[14.0][FIX] rma: add stock move in supplier group

DavidJForgeFlow committed:
[IMP] rma: cancel rma_lines

LoisRForgeFlow committed:
[FIX] rma: in multi step routes, only reserve first step

We shoul not force reservation on next steps on a multi step
route, oherwise a inconsistency is generated and the transfers
cannot be processed or cancel so the user gets stuck ("it is
not possible to unreserve more products that you have in stock"
error).

DavidJForgeFlow committed:
[IMP] rma: limit state statusbar

DavidJForgeFlow committed:
[FIX] rma: deprecated test warning fix

DavidJForgeFlow committed:
[16.0][FIX] rma: return of button done

DavidJForgeFlow committed:
[FIX] rma: product_uom_qty not in move_line_ids

DavidJForgeFlow committed:
[FIX] rma: make picking product_uom_qty

DavidJForgeFlow committed:
[FIX] rma: some fixups

BernatPForgeFlow committed:
[IMP] rma: Avoid error if group_id = False

SergiCForgeFlow committed:
[FIX] Restrict approval rights to RMA Manager

DavidJForgeFlow committed:
[IMP] rma: add date to rma_order_line

DavidJForgeFlow committed:
[FIX] rma: get all partner RMA

DavidJForgeFlow committed:
[FIX] rma: dropship from vendor as outgoing

DavidJForgeFlow committed:
[FIX] rma: get_move_rma dropship

JoanSForgeFlow committed:
[FIX] rma: procurement origin

In the current implementation of Odoo's _assign_picking() method in stock.move, there's a conditional check that looks at whether all the moves associated with a picking have the same partner_id and origin. If any move doesn't align with these conditions, the origin of the picking is set to False.

        if any(picking.partner_id.id != m.partner_id.id or
                picking.origin != m.origin for m in moves):
            # If a picking is found, we'll append `move` to its move list and thus its
            # `partner_id` and `ref` field will refer to multiple records. In this
            # case, we chose to  wipe them.
            picking.write({
                'partner_id': False,
                'origin': False,
            })
In the context of RMA when we have multiple moves associated with a picking, each coming from a different RMA order line, we encounter a problem. Each move has its origin set as the name of the RMA orde line (line.name), so as soon as a second move from a different line is appended to the picking, the origin of the picking is wiped, because it doesn't match the origin of the first move.

In order to prevent the partner_id of the picking from being set to False when there are multiple associated moves, I propose that we change the origin of the procurement from the name of the RMA line to the name of the procurement group (group.name). This way, all moves associated with a picking will share the same origin, preserving the origin of the picking and ensuring it doesn't get inadvertently set to False.

BernatPForgeFlow committed:
[IMP] rma: Auto-calculate package for serial products

When returning or delivering a serial product from a RMA, we will calculate by default its package. Later, it can be modyfied by the user.

ChrisOForgeFlow committed:
[FIX] quants can be more than one on _gather function result

JordiMForgeFlow committed:
[FIX] rma: correctly display fields in product category view

BernatPForgeFlow committed:
[FIX] rma: Do not set 'Destination Package' if RMA package comes from 'Customers' location

florian-dacosta committed:
rma : Fix description size in rma order line view form

AaronHForgeFlow committed:
[IMP] rma: date_rma in lines

ChrisOForgeFlow committed:
[14.0][IMP] added default operation on rma group, easy setup before rma lines created (#452)

* [14.0][IMP] added default operation on rma group, easy setup before rma lines created

* [IMP] added fields for default route created by wizard on rma group

* fix: get right price after create rma order line

AaronHForgeFlow committed:
[FIX] rma: inventory user by default should not see RMAs

AaronHForgeFlow committed:
[FIX] rma: RMA location in the warehouse should not be inside stock

JordiMForgeFlow committed:
[IMP] rma: mark RMA location as return location

florian-dacosta committed:
[FIX] Delete rma_picking_wizard_item on cascade to avoid failure

florian-dacosta committed:
[IMP] Remove domain from onchange method and put it in field view

ChrisOForgeFlow committed:
[14.0][FIX] rma, add support to handle package used on customer moves

DavidJForgeFlow committed:
[IMP] rma: add rma_line_id to stock.move views

When looking at stock moves created from a rma group, it may be confusing knowing who's move is for which RMA line. This improvement should avoid these confusions.

chafique-delli committed:
[16.0][IMP] rma: rma group state

chafique-delli committed:
[16.0][IMP] rma: create incoming shipment from rma group

chafique-delli committed:
[IMP] create delivery from rma group

chafique-delli committed:
[IMP] create supplier rma from rma group

AaronHForgeFlow committed:
[IMP] rma: black, isort, prettier

CarlosVForgeFlow committed:
[MIG] rma: Migration to v17

AaronHForgeFlow committed:
[FIX] rma: lot reservation error:
qty_done -> quantity
reserved_qty field does no exist anymore

chafique-delli committed:
[ADD] french translation for RMA module

DavidJForgeFlow committed:
[IMP] rma: add report menu

AaronHForgeFlow committed:
[FIX] rma: vendor rma lot mnmg

AaronHForgeFlow committed:
[IMP] rma: adapt icon to v17

florian-dacosta committed:
[IMP] rma : propagate cancelation
When canceling a rma order line, it will also cancel the previous (orig) steps.
Following steps (dest) can be managed with the propagate cancel option of the stock rules.
This commit also avoid canceling a whole picking which is problematic in case of the use of RMA groups

chafique-delli committed:
[FIX] rma, rma_account and rma_sale: following the comments that were made during the review

chafique-delli committed:
[IMP] rma, rma_account, rma_sale: move override of rma.order views from wizard views to rma_order_view.xml files

chafique-delli committed:
[ADD] smart button for sales in rma group

ChrisOForgeFlow committed:
[16.0][FIX] rma, change filter of moves to allow select pickings with children of partner selected

florian-dacosta committed:
[IMP] rma : add index on stock_move.rma_line_id

DavidJForgeFlow committed:
[FIX] rma: fix action that opens related RMA or RTV.

AaronHForgeFlow committed:
[FIX] rma: stock manager should not inherit rma manager permissions, that group must be explicity added

AlexPForgeFlow committed:
[FIX] rma: _check_move_partner compare with RMA commercial partner

florian-dacosta committed:
[FIX] rma: qty received and shipped in case of multiple step reception/expedition
Now the moves are filtered and only 1 move of the chain is selected to compute the quantity (the first one for reception and last one for expedition), so we should no longer ignore those to compute the quantities

florian-dacosta committed:
[FIX] rma : fix label and translations for incoming shipments

florian-dacosta committed:
[16][FIX] rma: make date_rma stored once again on rma_order

DavidJForgeFlow committed:
[IMP] rma: hide picking buttons when count == 0

JordiBForgeFlow committed:
[IMP] rma: Calculate expected date in pickings based on the customer lead time

DavidJForgeFlow committed:
[FIX] rma

Avoid reserving units that are not available. Also ensures the lot picked is the correct one by adding a dependency to stock_move_forced_lot.

JasminSForgeFlow committed:
[IMP] rma: update configuration view

LoisRForgeFlow commented:
@JasminSForgeFlow Is this https://github.com/ForgeFlow/stock-rma/pull/572 included here?

JasminSForgeFlow commented:
> @JasminSForgeFlow Is this #572 included here?

Yes, it was changed already

DavidJForgeFlow committed:
[REF] rma: change stock_move_forced_lot to stock_restrict_lot

The two modules have the same features, but the stock_restrict_lot is more used in other modules so changing the module used to match the OCA behavior.

florian-dacosta committed:
[FIX] Remove highlight on supplier rma button

florian-dacosta committed:
[FIX] Change attrs to invisible attribute in rma view

LoisRForgeFlow reviewed (commented):
Hi @JasminSForgeFlow 

An alignment of versions has been done in v17 https://github.com/ForgeFlow/stock-rma/pull/577. Could you take of aligning 17 and 18 again here?

Apart from that PR, there is potentially some others that were merged in 17 after this migration was created:

![image](https://github.com/user-attachments/assets/a5df466a-6446-4509-acf1-d9bd33284e58)

Please, make sure to leave anything behind.

cc @DavidJForgeFlow

JasminSForgeFlow committed:
[IMP] rma: black, isort, prettier

JasminSForgeFlow committed:
[MIG] rma: Migration to 18.0

DavidJForgeFlow commented:
Hi @JasminSForgeFlow could you also add https://github.com/ForgeFlow/stock-rma/pull/563, #550 and #555? `748d01939bcf6374605da1f3e95a6bd8e6b4f9e2`
`4ad1c3865e4e1417d6dead03c8c4be57f9b10765`
`28174e49a9018f429136eebd2a3815068624143b`

DavidJForgeFlow committed:
[FIX] rma: change invisible for column_invisible in tree views when trying to hide the full column

LoisRForgeFlow commented:
One more https://github.com/ForgeFlow/stock-rma/pull/590 :sweat_smile:

MeritxellAForgeFlow committed:
[14.0][IMP] rma: hide buttons when count is 0

LoisRForgeFlow commented:
@JasminSForgeFlow Dependency has been merged

DavidJForgeFlow commented:
Hi @JasminSForgeFlow ,
Can you fix the tests so we can start merging this? Thanks!

JasminSForgeFlow commented:
> Hi @JasminSForgeFlow , Can you fix the tests so we can start merging this? Thanks!

It has been fixed.
Thanks

DavidJForgeFlow commented on rma/data/stock_data.xml:
```diff
@@ -138,69 +138,4 @@
         <field name="rma_selectable" eval="True" />
     </record>
-
-    <record id="rule_rma_customer_in_pull" model="stock.rule">
```
Why did you remove all the rules?

LoisRForgeFlow replied:
Because they were duplicated, as they were generated also by code

DavidJForgeFlow replied:
Makes sense!

DavidJForgeFlow commented on rma/models/stock_warehouse.py:
```diff
@@ -69,7 +69,6 @@ def write(self, vals):
                                 "usage": "internal",
                                 "location_id": wh.view_location_id.id,
                                 "company_id": wh.company_id.id,
-                                "return_location": True,
```
Why this change? The RMA location isn't by definition a return location?

JasminSForgeFlow replied:
Field has been removed in 18.0

DavidJForgeFlow replied:
Makes sense!

DavidJForgeFlow commented on rma/models/stock_move.py:
```diff
@@ -0,0 +1,123 @@
+            return self.location_id.usage
+
+    @api.model
+    def _get_last_usage(self):
```
maybe here we should take into account the new location_final_id field.

JasminSForgeFlow replied:
I have tested it, but it's not working in all cases.
test_11, deliver RMA does not work with that change.

JasminSForgeFlow replied:
My bad, it's working.
I have updated it.

Thanks

DavidJForgeFlow replied:
I think it should look like:
```
if self.move_dest_ids:
    # We assume here that all origin moves come from the same place
    return self.move_dest_ids[0]._get_last_usage()
else:
    return self.location_final_id.usage if self.location_final_id else self.location_dest_id.usage
```

JasminSForgeFlow replied:
Updated as per suggestion.
Thanks

JasminSForgeFlow committed:
[IMP] rma: Fix calculation for qty_to_receive

DavidJForgeFlow commented:
@AaronHForgeFlow IMO this is ready to merge, what do you think?

AaronHForgeFlow commented on rma/models/procurement_group.py:
```diff
@@ -0,0 +1,37 @@
+        if rma_route_check:
+            if res and not res.route_id.rma_selectable:
+                raise ValidationError(
+                    _(
```
```suggestion
                    self.env._(
```

Can you change all occurrences?

JasminSForgeFlow replied:
Done, Thanks

AaronHForgeFlow commented on rma/views/rma_operation_view.xml:
```diff
@@ -0,0 +1,88 @@
+                        name="web_ribbon"
+                        title="Archived"
+                        bg_color="bg-danger"
+                        invisible="active == True"
```
```suggestion
                        invisible="active"
```

JasminSForgeFlow replied:
Done, Thanks

AaronHForgeFlow commented on rma/views/rma_operation_view.xml:
```diff
@@ -0,0 +1,88 @@
+                    <div class="oe_button_box" name="button_box" />
+                    <group colspan="4" col="2">
+                        <group name="description" string="Description">
+                            <field name="active" invisible="1" />
```
```suggestion
```

JasminSForgeFlow replied:
Done, Thanks

AaronHForgeFlow commented on rma/views/rma_order_line_view.xml:
```diff
@@ -0,0 +1,539 @@
+                                <field name="date_rma" />
+                                <field name="requested_by" readonly="1" />
+                                <field name="assigned_to" />
+                                <field name="type" invisible="1" />
```
I think having this field with invisible here only to be used in attributes is not necessary anymore

JasminSForgeFlow replied:
Done, Thanks

AaronHForgeFlow reviewed (approved):
LGTM! :+1:  just made some minor comments

ThiagoMForgeFlow commented on rma/models/stock_warehouse.py:
```diff
@@ -0,0 +1,269 @@
+        return True
+
+    def write(self, vals):
+        if "rma_in_this_wh" in vals:
```
As stock.rules are created from the values of this field, it could be a good idea to inherit the create method and apply the same logic.

AaronHForgeFlow commented:
@JasminSForgeFlow  I think you can also merge this one https://github.com/ForgeFlow/stock-rma/pull/624, I think it is a nice improvement.

JasminSForgeFlow committed:
Merge pull request #624 from akretion/18.0-mig-rma-imp-group

[18][rma] Add possibility to create incoming/delivery orders and customer/supplier rma from rma group

JasminSForgeFlow commented:
> @JasminSForgeFlow I think you can also merge this one #624, I think it is a nice improvement.

Done

LoisRForgeFlow reviewed (approved):
I have been blocking this for too long. I have tested and overall it works good and as a migration is more than ready, I'll merge and create issues for the details that are pending.
